### PR TITLE
10 feature backend volume implementation

### PR DESCRIPTION
**Pull request by Alexvc23**
**As a developer**
- *what*:
  - **backend volume**
    - I want to implement a `backend` **volume** bind to the local host.
  - **devDependencies**
    - I installed all the `devDependencies` 
  - **readMe.md files** 
    - Update all readMe.md files
- *how*:
  -  **backend volume**
    - Applying changes in docker-compose file
  - **devDependencies**
    - I added `nodemon` and `ts-node` *dependencies* in `package.json`. Lastly, I created a `nodemon.json` file.
  - **readMe.md**
    - modifying directly the `readMe.md` files for the entire project, `frontend` and `backend`.
- *why*
  - **backend volume**
    -  So we do modification to the `backend` and see changes in the browser on the fly, without having to relaunch docker-compose.
  - **devDependencies**
    - `nodemon` watches if there are modifications in the `.ts` files to restart the `backend`
    - `ts-node` compiles `.ts` code into `.js` code
    - `nodemon.json` file configures `nodemon` dependencie and target the files to watch. 
  - **readMe.md files**
    - Is important to have a basic understanding of the project and its structure, the readMe.md will clarify all the terminology, the purpose of each file, and provide instructions. It is an important guidance.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -22,9 +22,11 @@ services:
     env_file:
       - ./.env # Set the environment file
```
Comment (Noushhhh) [inline docker-compose.yml:23]: .env ?

File: srcs/backend/readMe.md [hunk 2]
```diff
@@ -23,57 +23,52 @@ This is the backend application for the project.
...
-- "reflect-metadata": "^0.1.13"
+- "reflect-metadata"
   - ✨ Utility: Provides a way to add and read metadata to/from TypeScript classes and their properties.
 
-- "rxjs": "^7.8.1"
+- "rxjs": "7.8.1"
   - ✨ Utility: Reactive Extensions for JavaScript (RxJS) - Library for composing asynchronous and event-based programs.
 
-- "serve-static": "^1.14.1"
+- "serve-static"
```
Comment (Noushhhh) [inline srcs/backend/readMe.md:53]: no version suggested ?

File: srcs/backend/src/app.service.ts [hunk 3]
```diff
@@ -3,6 +3,6 @@ import { Injectable } from '@nestjs/common';
 @Injectable() // Declare this class as an injectable service
 export class AppService {
   getHello(): string {
-    return 'Hello World!';
+    return 'ft_transcendance with the best team!';
```
Comment (Noushhhh) [inline srcs/backend/src/app.service.ts:6]: cutyyyy pieeee hihi

**Review by Noushhhh — COMMENTED** [review]
Perfect Alex great Job. Still have a problem when trying to run the app on my MacBook. We ll see if some update can be loaded using the entry point ? Just a path for .env file to update as you ll see in the comments. Thank you for your rigour and investment.

File: docker-compose.yml (hunk 1 above)
Comment (Alexvc23) [reply docker-compose.yml:23]: as you like, that change anything 😃

**Comment by Alexvc23** [comment]
> Perfect Alex great Job. Still have a problem when trying to run the app on my MacBook. We ll see if some update can be loaded using the entry point ? Just a path for .env file to update as you ll see in the comments. Thank you for your rigour and investment.

@Noushhhh can you please try to run your app using my branch, just to see what happens ?

**Pull request merged**